Add note about index.html routing (#972)
zeit/docs: pages/docs/v2/routing/directory-listing.mdx

## Index Files

>  **Index Files** and rendered in place of the path of the directory that they are located.

Reviewer: ```suggestion
 **Index Files** are rendered in place of the path of the directory they are located in.
```

Author: Fixed

> For example, visiting the path `/about` will serve the `/about/index.html` file.

Reviewer: If it exists? I'm not sure if saying this will make it sound like it always explicitly looks for `index.html` for directories like that 🤔

Author: Fixed, thanks!

> Files that match all of the following criteria will be used:

Reviewer: ```suggestion
Files that match all of the following criteria will be identified as an index:
```

Author: Fixed

- The file name is either `index` or `index.<extension>` (where `<extension>` is any file extension of your choice).
- The file name **does not** end with a `.` or contain multiple `.` in a row.

> If multiple files match this pattern, `index.html` will takes precedence followed by shortest to longest file names. That means `index.js` will take precedence over `index.bundle.js`.

Reviewer: ```suggestion
If multiple files match this pattern, `index.html` will take precedence followed by shortest to longest file names. That means `index.js` will take precedence over `index.bundle.js`.
```

Also, I think we can explain this better, because it's assuming `index.html` exists in the content and is off-putting, I think.

Maybe
```suggestion
If multiple files match this pattern, if a `index.html` within the list, it will take precedence, otherwise (and followed by), the shortest to longest file names, e.g., `index.js` will take precedence over `index.bundle.js`.
```

Author: This revision is very confusing 🤔 

I think it was better the way I wrote it 😄

Reviewer: Fair enough! Let's do it

Author: I'm using the word `precedence`, it doesn't have to exist. If it does it has precedence over other matches.
